### DISCUSSION TITLE
AttributeError: 'str' object has no attribute 'loop'

### DISCUSSION THREAD

**Discussion by codingMASTER398** [Rapptz/discord.py#6087] [Q&A]
I have used this for a while, but moving to pythonanywhere gave me this error:

_AttributeError: 'str' object has no attribute 'loop'_

this is my code:

```py
import requests
import time

import discord
import os

class MyClient(discord.Client):
    async def on_ready(self):
        print('Logged on as {0}!'.format(self.user))
        while True:
          channel = client.get_channel(781385928937701376)
          try:
            response = requests.get('http://kashooting.tech')
            print(int(response.status_code))
            if response.status_code == 200:
              await channel.send('Server status is currently okay. Pinging took '+response.elapsed.total_seconds())
            else:
              await channel.send('Something went wrong while contacting the server! Something is not right.')

          except:
            await channel.send('Something went wrong while contacting the server! Something is not right.')


          time.sleep(100)


MyClient.run('TOKEN')
```

**Is there anything i can do?**

**Comment by oliver-ni** [comment]
> ```python
> MyClient.run('TOKEN')
> ```

Learn the difference between classes and instances. You should be making an instance of `MyClient` and calling run on that.

Besides that, there are a lot of other issues with your code...

**Comment by codingMASTER398** [comment]
How would i make an instance out of a class?
Im not the best at python and usually use node.js, so the code is not great

**Comment by Rapptz** [comment]
There are too many issues with this code.

1. Your loop uses `client.get_channel` but `client` is not defined anywhere. This will lead to a `NameError`. You probably meant to use `self` here, since you refer to your own class.
2. You're using `requests` and `time.sleep`, both of which are [blocking code](https://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean) and is inadvisable. Consider using `aiohttp` and `asyncio.sleep` instead.
3. `MyClient.run('TOKEN')` uses the class rather than an instance, which is generally a basic Python issue. In order to make an instance of a class you have to create it using the constructor, denoted by parentheses (e.g. `MyClient()`). 

If you're having trouble with Python it is advisable to learn the fundamentals before attempting to make a bot. This repository isn't really the best place to get beginner help with Python as you're expected to know the fundamentals. Consider consulting a tutorial such as [the official Python.org tutorial](https://docs.python.org/3/tutorial/) or one of the many resources online.

**Reply by codingMASTER398** [reply]
Thanks!

**Comment by pikaninja** [comment]
we could create one by doing something like 
```py
myinstance = MyClient()
```
which would create an instance of MyClient

**Reply by codingMASTER398** [reply]
When i do that i get a 

aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host discord.com:443 ssl:default [Connect call failed ('162.159.136.232', 443)]

error instead. D:
